### PR TITLE
[quant] Input-Weight Equalization - Conv prepare support

### PR DESCRIPTION
**Pull request by angelayi**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #61350
* #61287
* **#61286**
* #61285

Summary: Modifies the prepare step to support conv layers during input-weight equalization and adds tests to make sure that the results are as expected.

Initial: 
```
      w
      |
x -> conv -> y
```

After prepare:

```
                                         w
                                         |
                                  weight_quant_obs
                                         |
                                    weight_eq_obs
                                         |
x -> input_quant_obs -> input_eq_obs -> conv -> out_quant_obs -> y
```

Test Plan: `python test/test_quantization.py TestEqualizeFx.test_input_weight_equalization_prepare`

Initial:
```
ConvModel(
  (conv): Conv2d(3, 5, kernel_size=(3, 3), stride=(1, 1), bias=False)
)
```

After prepare:
```
graph():
    %x : [#users=1] = placeholder[target=x]
    %x_activation_post_process_0 : [#users=1] = call_module[target=x_activation_post_process_0](args = (%x,), kwargs = {})
    %x_activation_post_process_0_equalization_process_0 : [#users=1] = call_module[target=x_activation_post_process_0_equalization_process_0](args = (%x_activation_post_process_0,), kwargs = {})
    %conv : [#users=1] = call_module[target=conv](args = (%x_activation_post_process_0_equalization_process_0,), kwargs = {})
    %conv_activation_post_process_0 : [#users=1] = call_module[target=conv_activation_post_process_0](args = (%conv,), kwargs = {})
    return conv_activation_post_process_0
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D29557051](https://our.internmc.facebook.com/intern/diff/D29557051)

### CODE REVIEW

**Comment by angelayi** [comment]
@angelayi has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29557051).

**Review by supriyar — APPROVED** [review]
test file changes look good

**Comment by angelayi** [comment]
@angelayi has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29557051).

File: torch/quantization/fx/_equalize.py [hunk 1]
```diff
@@ -211,13 +211,14 @@ def __new__(cls, input_activation=torch.nn.Identity, weight=torch.nn.Identity):
 
 def node_supports_equalization(node: Node, modules) -> bool:
     """ Checks if the current node supports equalization
-    Currently we only support nn.Linear and F.Linear layers
+    Currently we only support nn.Linear/F.Linear and nn.Conv/F.conv layers
     """
     if node.op == 'call_module':
         return (isinstance(modules[node.target], nn.Linear) or
-                isinstance(modules[node.target], nni.LinearReLU))
+                isinstance(modules[node.target], nni.LinearReLU) or
```
Comment (jerryzh168) [inline torch/quantization/fx/_equalize.py:218]: nit: we can use type(modules[node.target]) in [nn.Linear, nni.LinearReLU, nn.Conv2d]

File: torch/quantization/fx/_equalize.py [hunk 2]
```diff
@@ -211,13 +211,14 @@ def __new__(cls, input_activation=torch.nn.Identity, weight=torch.nn.Identity):
...
+                isinstance(modules[node.target], nn.Conv2d))
     elif node.op == 'call_function':
-        return node.target == F.linear
+        return node.target == F.linear or node.target == F.conv2d
```
Comment (jerryzh168) [inline torch/quantization/fx/_equalize.py:221]: same here

**Review by jerryzh168 — APPROVED** [review]
LGTM!

**Comment by angelayi** [comment]
@angelayi has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29557051).

**Pull request closed without merging**